### PR TITLE
Add tailwindcss

### LINKED ISSUE #5: Enable Tailwind in this project
Tailwind needs to be accessible to this project. 

https://daily-dev-tips.com/posts/how-to-use-tailwind-css-in-astro/

```
npm install -D tailwindcss
```

there are some follow up config files needed.

```js
// tailwind.config.js

module.exports = {
  mode: 'jit',
  purge: ['./public/**/*.html', './src/**/*.{astro,js,jsx,ts,tsx,vue}'],
};
```

```js
// astro.config.js
devOptions: {
  tailwindConfig: './tailwind.config.js';
}
```

```css
// global.css 

@tailwind base;
@tailwind components;
@tailwind utilities;
```

### LINKED ISSUE #6: Create Repo card components
## Today the site content looks like this when displaying GitHub repos. 

Screen Shot 2021-12-15 at 6 58 30 AM

## What we would like it to look like 

Screen Shot 2021-12-15 at 6 55 01 AM

Steps to implement:

1. Apply Tailwind UI to a reusable React component
2. Add the component to the render repo

Link to CSS: 

1. [PostGrid](https://github.com/bdougie/hot-open-sauced/blob/main/components/post/PostGrid.jsx)

### LINKED ISSUE #7: Create PostList components
## Today the site content looks like this when displaying GitHub repos. 

Screen Shot 2021-12-15 at 6 58 30 AM

## What we would like it to look like 

Screen Shot 2021-12-15 at 7 00 19 AM

Steps to implement:

1. Apply Tailwind UI to a reusable React component
2. Add the component to the render repo

Link to CSS: 

1. [PostList](https://github.com/bdougie/hot-open-sauced/blob/main/components/post/PostList.jsx)

### LINKED ISSUE #10: Implement the weekly goal modal
Need to implement the encouragement modal. This is will be for display only, the logic for this will be implement in the future.

Screen Shot 2021-12-15 at 7 07 02 AM
 
Link to elements:
1. [Code](https://github.com/bdougie/hot-open-sauced/blob/main/components/layout/Modal.jsx)
2. [Live App](https://hot-open-sauced.vercel.app/)

### PR DESCRIPTION
**Pull request by JScrips**
## What type of PR is this? (check all applicable)

- [ ] ♻️ Refactor
- [x] ✨ Feature
- [x] 🐛 Bug Fix
- [ ] 👷 Optimization
- [ ] 📝 Documentation Update
- [ ] 🔖 Release
- [ ] 🚩 Other

## Description

This PR adds Tailwind and some basic UI component structure. 

## Related Tickets & Documents

[Enable Tailwind in this project ]-- closes #5
[ Create Repo card components ]-- closes #6
[ Create PostList components ]--closes #7
[ PostList and PostGrid toggle and nav bar ]--(https://github.com/open-sauced/hot-sauce/issues/8)
[ Add navigation/filter ]-- (https://github.com/open-sauced/hot-sauce/issues/9)
[ Implement the weekly goal modal ]--closes #10

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 readme
- [ ] 📜 contributing.md
- [ ] 📓 docs
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

## [optional] Are there any post-deployment tasks we need to perform?
NO

## [optional] What gif best describes this PR or how it makes you feel?
This is my first public PR.

### CODE REVIEW

File: postcss.config.cjs [hunk 1]
```diff
@@ -0,0 +1,4 @@
+// postcss.config.cjs
+module.exports = {
+    plugins: [require('tailwindcss')],
+  };
```
Comment (bdougie) [inline postcss.config.cjs:4]:
```suggestion
    plugins: [require('tailwindcss')],
  };
```
```suggestion
  plugins: [require('tailwindcss')],
};
```

**Comment by bdougie** [comment]
@JScrips thanks for turning this around so quickly. There are some small updates needed to make it right. 

FYI, the checkboxes can be clicked. 

This
- [ x] 🙅 no, because they aren't needed

Should be (you added an extra space in there

- [x] 🙅 no, because they aren't needed

File: src/components/Modal.jsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+                            </div>
+                        </div> 
+                    </div>
+                </div>
+            </div>
+        </div>
+        </>
+        
+    )
+}
```
Comment (bdougie) [inline src/components/Modal.jsx:71]: Spacing in this file seems to be Tabbed or 4 spaces. Can you be updated to 2 spaces similar (for consistency) to src/components/LayoutToggle.jsx

File: src/components/Navbar.astro [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+</div>
+
+<div class=" rounded-full w-10 h-10 overflow-hidden ring-2 ring-red-800 ">
+                <img
+                    class="object-cover w-[500] h-[500]"
+                    src={av1}
+                    alt="Avatar 02"
+                />
+</div>
+</nav>
```
Comment (bdougie) [inline src/components/Navbar.astro:18]: Spacing is all over here. Can we update to 2 spaces

File: src/components/Navbar2.astro [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+ <div class=" bg-darkestGrey py-16 md:py-20">
+            <nav class=" container">
+                <ul class="flex flex-col sm:flex-row text-xl font-righteous text-accent font-bold items-center cursor-pointer  ">
+                    <li class=" bg-cheesyYellow px-2 py-2 rounded-xl text-grey hover:text-saucyRed mr-3 sm:mr-11 " >Popular</li>
+                    <li class=" hover:text-saucyRed transition-all duration-300 mr-3 sm:mr-11 ">Upvoted</li>
+                    <li class="hover:text-saucyRed transition-all duration-300 mr-3 sm:mr-11 ">Discussed</li>
+                    <li class=" hover:text-saucyRed  transition-all duration-300 mr-3 sm:mr-11 ">Recent</li>
+                </ul>
+            </nav>
+</div>
```
Comment (bdougie) [inline src/components/Navbar2.astro:12]: 2 spaces

**Review by bdougie — CHANGES_REQUESTED** [review]
Some small spacing fixes are needed.

**Comment by bdougie** [comment]
A few more things need to be addressed, before considering these issues complete.

- [ ] Description in the PR needs to be updated to not include placeholder text
- [ ] None of the font-awesome icons transferred over.

**Review by kelvinsekx — COMMENTED** [review]
Also there is a need for an explicit Layout component. That is then imported into the index.astro. it makes the work cleaner that way.

File: src/components/Modal.jsx (hunk 2 above)
Comment (kelvinsekx) [reply src/components/Modal.jsx:71]: Just saying, what if we implement .husky to enforce linting?

**Comment by bdougie** [comment]
> Also there is a need for an explicit Layout component. That is then imported into the index.astro. it makes the work cleaner that way.

Can you start that on another branch @kelvinsekx. I am going to merge this to unblock folks, but I can't call #8, #9, and #10 closed, since they are missing font-awesome elements. 

@JScrips feel free to open a second PR to fix the spacing, and in the future open 1 PR per issue. That way we can avoid assignment confusion.

**Review by bdougie — APPROVED** [review]
Approving to unblock folks. Added #14 to address the spacing in the future.

**Pull request merged**